viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        print("hello")
+        dict = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return dict
+
+    return None
+
+def add_to_watched(user_data, movie):
+    user_data_copy = user_data.copy()
+
+    # if movie:
+    user_data_copy["watched"].append(movie)
+
+    return user_data_copy
+
+def add_to_watchlist(user_data, movie):
+    user_data_copy = user_data.copy()
+
+    user_data_copy["watchlist"].append(movie)
+
+    return user_data_copy
+
+def watch_movie(user_data, title):
+    # user_data_copy = user_data.copy()
+    for movie_entry in user_data["watchlist"]:
+        if movie_entry["title"] == title:
+            user_data["watched"].append(movie_entry)
+            user_data["watchlist"].remove(movie_entry)
+

```

Comment:
Here I might suggest that you actually add a break in your if block.

Note that, while we might only expect there to be one movie with a particular title in the watchlist (might not be a good assumption, but for this simplified data, it might be ok), this implementation is fine.

As soon as we make a change (like removing from watchlist) we want to stop iterating, so we use break. By adding break after line 35, the loop iterates the watchlist, and when the desired movie is found, adds it to the watched list and removes it from the watchlist. If we returned immediately at this point, everything would be fine!

Without the break statement, we might be in a scenario where title actually matches to more than one movie as it loops (without the break) and we'll see some weirdness.